### DISCUSSION TITLE
Able to Query but not Retrieve - Refused: OutOfResourcesSubOperations

### DISCUSSION THREAD

**Discussion by elvisshock** [dcm4che/dcm4chee-arc-light#3206] [Q&A]
Hi Team,

There seems to be a problem with dcm4chee arc light. I have tried different versions (including 5.5.1), but keep running into the same problem - I am able to query, but I am not able to retrieve the dicom files using C-GET. I have tried googling for this error, but everyone seems to be having some sort of network issue. I think I have eliminated for that. Here's what I have done:
1. Setup my server on docker using the instructions provided on https://github.com/dcm4che/dcm4chee-arc-light/wiki/Running-on-Docker. System Setup summary:
   Mac OS X - El Capitan
   Docker 1.12.0-rc3-beta18
   dcm4che/dcm4chee-arc-psql:5.5.1-secure-ui
   dcm4che/postgres-dcm4chee:5.5
   dcm4che/slapd-dcm4chee:5.5.1
2. I am able to push files to the server. I am using Osirix.
3. I am able to query the records on the server correctly. (again, using Osirix)
4. But when I try to get a file by double clicking on the record in Osirix, it throws an error - "Get Failed: OutOfResourcesSubOperations". 
5. I tried the setup with settings from my public IP as well as the localhost. Both are behaving the same way, so the error doesn't seem to be from the firewall or the router. 
6. I tried setting up the server on docker using previous versions (5.5.0), but run into the same issue.
7. Lastly, I also setup a server on a Virtual Box and dcm4chee - and the server seems to be working fine with the dcm4chee-2.18.1-mysql setup. Both Query and Retrieve are working with that - on public and localhost.

I am not sure if this is a bug or I am missing something. Would appreciate any help. 

Thanks!

**Comment by gunterze** [comment]
Cannot reproduce the problem using dcm4che-3 getscu utility:

```
$ getscu -c DCM4CHEE@localhost:11112 -mStudyInstanceUID=2.16.840.1.113662.5.1335979110.1.1052340890
10:59:47,059 INFO  - Initiate connection from 0.0.0.0/0.0.0.0:0 to localhost:11112
10:59:47,064 INFO  - Established connection Socket[addr=localhost/127.0.0.1,port=11112,localport=41358]
10:59:47,070 INFO  - GETSCU->DCM4CHEE(1) << A-ASSOCIATE-RQ
10:59:47,076 INFO  - GETSCU->DCM4CHEE(1) >> A-ASSOCIATE-AC
10:59:47,081 INFO  - GETSCU->DCM4CHEE(1) << 1:C-GET-RQ[pcid=1, prior=0
  cuid=1.2.840.10008.5.1.4.1.2.2.3 - Study Root Query/Retrieve Information Model - GET
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian
10:59:47,096 INFO  - GETSCU->DCM4CHEE(1) >> 1:C-GET-RSP[pcid=1, remaining=1, completed=0, failed=0, warning=0, status=ff00H
  cuid=1.2.840.10008.5.1.4.1.2.2.3 - Study Root Query/Retrieve Information Model - GET
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian
10:59:47,097 INFO  - GETSCU->DCM4CHEE(1) >> 1:C-STORE-RQ[pcid=83, prior=0
  cuid=1.2.840.10008.5.1.4.1.1.20 - Nuclear Medicine Image Storage
  iuid=2.16.840.1.113662.5.1335979110.3.14.222.11.959118019 - ?
  tsuid=1.2.840.10008.1.2.1 - Explicit VR Little Endian
10:59:47,097 INFO  - GETSCU->DCM4CHEE(1): M-WRITE ./2.16.840.1.113662.5.1335979110.3.14.222.11.959118019
10:59:47,099 INFO  - GETSCU->DCM4CHEE(1) << 1:C-STORE-RSP[pcid=83, status=0H
  cuid=1.2.840.10008.5.1.4.1.1.20 - Nuclear Medicine Image Storage
  iuid=2.16.840.1.113662.5.1335979110.3.14.222.11.959118019 - ?
  tsuid=1.2.840.10008.1.2.1 - Explicit VR Little Endian
10:59:47,102 INFO  - GETSCU->DCM4CHEE(1) >> 1:C-GET-RSP[pcid=1, completed=1, failed=0, warning=0, status=0H
  cuid=1.2.840.10008.5.1.4.1.2.2.3 - Study Root Query/Retrieve Information Model - GET
  tsuid=1.2.840.10008.1.2 - Implicit VR Little Endian
10:59:47,102 INFO  - GETSCU->DCM4CHEE(1) << A-RELEASE-RQ
10:59:47,103 INFO  - GETSCU->DCM4CHEE(1) >> A-RELEASE-RP
10:59:47,103 INFO  - GETSCU->DCM4CHEE(1): close Socket[addr=localhost/127.0.0.1,port=11112,localport=41358]
```

Check the exception Stack in Wildfly server.log

**Comment by elvisshock** [comment]
Thanks for the tip. I tried the same thing - getting the dicom file using getscu and it worked successfully as well. However, Osirix is still erroring out. The Wildfly server.log is also clean.

I tried accessing the images through a different DICOM Viewer, and I am able to successfully download them. I am guessing it's something wrong with Osirix. But I am not sure what's the issue with it. Will report it over there. Thanks!

One last thing, I wanted to try the WADO protocol to attempt to download the studies. Is there a default username password for WADO? While I found the URL, I couldn't find the username password in the wiki pages.

**Comment by gunterze** [comment]
Secured version of the RESTful services (as the secured version of the Web UI) uses OpenID Connect based on Auth 2.0 for authentication - not just http basic authentication!

Please post further question to https://groups.google.com/forum/#!forum/dcm4che

**Comment by luizinho44** [comment]
In Osirix, change the sintax of your listener to Explicit Little Indian.
